hardware_interface/include/hardware_interface/hardware_info.hpp

```diff
@@ -55,6 +65,8 @@ struct ComponentInfo
   /// Type of the component: sensor, joint, or GPIO.
   std::string type;
 
+  ///  If the component has a mimic joint tag, for opt-out
+  std::optional<bool> is_mimic{};
```

Comment:
Is there a reason for using `optional` here? as we already know definitively if the joint is mimic or not directly from URDF

Reply:
I found it an elegant solution of parsing the "optional" xml attribute. I have these cases I want to consider during URDF parsing:

- mimic attribute exists
   - is true: check for mimic info in URDF and add mimic_joints. throw an error if no mimic info.
   - is false: do nothing
- mimic attribute does not exist:
   - if joint has an mimic attribute, add it to mimic_joints
   - if joint has no mimic attribute, do nothing.

This can be done with the `std::optional`, otherwise I need two bools `mimic_exists` `mimic_value`. Or I loose the existence check if mimic=true but no mimic info exists.

Reply:
IMO, using optional with bool is something not recommended for various reasons as it is very easy to make mistakes. The problem specifically is `std::optional<T>::operator bool()`, which is easy to make mistakes and use mimic directly in the conditions. I usually avoid it for such reasons. I might be wrong as well.

As per your above approach, why not do the opposite, like check for the mimic joints in the URDF and update the components `is_mimic` variable and if a `mimic` attribute is defined for a joint in the ros2_control tag, but then if the `is_mimic` variable is false, should throw an error as it is not a `mimic` joint.

What do you think about it?

Reply:
Instead of `std::optional` I could also use a `std::string` for `is_mimic` with the three states:
- empty
- "true"
- "false"
And evaluate it inside of `parse_control_resources_from_urdf`
Just thought that `std::optional` is more declarative, but maybe prone for errors in this usage.

Reply:
maybe we can also skip the check for the case of attribute `mimic=true` for joints without mimic info. then a simple `bool` is sufficient.

Reply:
Yes exactly, that's what I meant is earlier, and also during the WG meeting, we agreed that the mimic has to be defined in the URDF joints definition for it to have the mimic functionality from the `ros2_control` tag. 

Regarding the debug output, you can do it where you are checking the mimic attribute, once it is true, you can verify it with with the URDF one and print something if they are not as expected. 

What do you think?

@bmagyar @destogl what's your opinion on this `optional<bool>`, do you think it is fine? or shall we change it to what we are discussing here?

Reply:
OK, if I'd parse the URDF before the ros2_control tags, and after checking for the mimic attribute I could traverse through the mimic_joints vector: in case of mimic=false the info gets removed from the vector. then I don't need a `is_mimic` variable at all.

Reply:
Sounds good!

Reply:
I replaced it with an enum. Otherwise the logic became too complicated.